### PR TITLE
[Feature] Add model ensemble tools

### PR DESCRIPTION
**Pull request by zhijiejia**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Convenient segmentation probability ensemble of multiple models

## Modification

Added `model_ensemble.py` files, added `aug_test_logits` and `simple_test_logits` functions, in `mmseg/models/segmentors/encoder_decoder.py`

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMDet3D.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Review by MeowZheng — COMMENTED** [review]
It looks good to me.

There is the last requirement that please add instructions in https://github.com/open-mmlab/mmsegmentation/blob/master/docs/en/useful_tools.md

and its Chinese version 
https://github.com/open-mmlab/mmsegmentation/blob/master/docs/zh_cn/useful_tools.md

**Comment by zhijiejia** [comment]
> It looks good to me.
> 
> There is the last requirement that please add instructions in https://github.com/open-mmlab/mmsegmentation/blob/master/docs/en/useful_tools.md
> 
> and its Chinese version https://github.com/open-mmlab/mmsegmentation/blob/master/docs/zh_cn/useful_tools.md

Thanks your help, I have added the zh_cn and en instruction in `docs/zh_cn/useful_tools.md` and `docs/en/useful_tools.md`

File: docs/en/useful_tools.md [hunk 1]
```diff
@@ -424,3 +424,26 @@ result/pred_result.pkl \
 result/confusion_matrix \
 --show
 ```
+
+## Model ensemble
+
+To complete the integration of prediction probabilities for multiple models, we provide 'tools/model_ensemble.py'
```
Comment (MeowZheng) [inline docs/en/useful_tools.md:430]: Could you add some description about the result after running this script?

File: tools/model_ensemble.py [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    parser.add_argument(
+        '--checkpoint',
+        type=str,
+        nargs='+',
+        help='ensemble checkpoint files path')
+    parser.add_argument(
+        '--aug-test',
+        action='store_true',
+        help='control ensemble aug-result or single-result (default)')
+    parser.add_argument('--out', help='the dir to save result')
```
Comment (MeowZheng) [inline tools/model_ensemble.py:106]: Add default value for out?

Comment (zhijiejia) [reply tools/model_ensemble.py]: Thanks for your comment, I added the default value of the --out parameter and added an introduction to the output of the model ensemble

File: tools/model_ensemble.py [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+            result.append(logits)
+
+        result_logits = 0
+        for logit in result:
+            result_logits += logit
+
+        pred = result_logits.argmax(axis=1).squeeze()
+        img_info = dataset.img_infos[batch_indices[0]]
+        file_name = os.path.join(tmpdir,
+                                 img_info['ann']['seg_map'].split('/')[-1])
```
Comment (MeowZheng) [inline tools/model_ensemble.py:87]:
```suggestion
                                 img_info['ann']['seg_map'].split(os.path.sep)[-1])
```

File: tools/model_ensemble.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+        '--checkpoint',
+        type=str,
+        nargs='+',
+        help='ensemble checkpoint files path')
+    parser.add_argument(
+        '--aug-test',
+        action='store_true',
+        help='control ensemble aug-result or single-result (default)')
+    parser.add_argument(
+        '--out', type=str, default='results/', help='the dir to save result')
```
Comment (MeowZheng) [inline tools/model_ensemble.py:107]:
```suggestion
        '--out', type=str, default='results', help='the dir to save result')
```

File: tools/model_ensemble.py [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+        cfg.model.pretrained = None
+        cfg.data.test.test_mode = True
+
+        model = build_segmentor(cfg.model, test_cfg=cfg.get('test_cfg'))
+        if cfg.get('fp16', None):
+            wrap_fp16_model(model)
+        load_checkpoint(model, ckpt, map_location='cpu')
+        torch.cuda.empty_cache()
+        eval_kwargs = {'imgfile_prefix': args.out}  # result save dir
+        tmpdir = eval_kwargs['imgfile_prefix']
```
Comment (MeowZheng) [inline tools/model_ensemble.py:59]:
```suggestion
        tmpdir = args.out
```

Comment (zhijiejia) [reply tools/model_ensemble.py]: Thanks for your suggestion, I have modified the relevant code.

**Review by MeowZheng — APPROVED** [review]
